signing/multisig/blsMultisig.go

```diff
@@ -1,332 +1,82 @@
 		return nil, crypto.ErrNilKeyGenerator
 	}
-	if ownIndex >= uint16(len(pubKeys)) {
-		return nil, crypto.ErrIndexOutOfBounds
-	}
-
-	sizeConsensus := uint16(len(pubKeys))
-	sigShares := make([][]byte, sizeConsensus)
-	pk, err := convertStringsToPubKeys(pubKeys, keyGen)
-
-	if err != nil {
-		return nil, err
-	}
-
-	data := &blsMultiSigData{
-		pubKeys:   pk,
-		privKey:   privKey,
-		ownIndex:  ownIndex,
-		sigShares: sigShares,
-	}
-
-	// own index is used only for signing
 	return &blsMultiSigner{
-		data:       data,
-		mutSigData: sync.RWMutex{},
-		keyGen:     keyGen,
-		llSigner:   llSigner,
+		keyGen:   keyGen,
+		llSigner: llSigner,
 	}, nil
 }
 
-// Reset resets the multiSigData inside the multiSigner
-func (bms *blsMultiSigner) Reset(pubKeys []string, index uint16) error {
-	if pubKeys == nil {
-		return crypto.ErrNilPublicKeys
-	}
-
-	if index >= uint16(len(pubKeys)) {
-		return crypto.ErrIndexOutOfBounds
-	}
-
-	sizeConsensus := uint16(len(pubKeys))
-	sigShares := make([][]byte, sizeConsensus)
-	pk, err := convertStringsToPubKeys(pubKeys, bms.keyGen)
-
-	if err != nil {
-		return err
-	}
-
-	bms.mutSigData.Lock()
-	defer bms.mutSigData.Unlock()
-
-	privKey := bms.data.privKey
-
-	data := &blsMultiSigData{
-		pubKeys:   pk,
-		privKey:   privKey,
-		ownIndex:  index,
-		sigShares: sigShares,
-	}
-
-	bms.data = data
-
-	return nil
-}
-
-// Create generates a multiSigner and initializes corresponding fields with the given params
-func (bms *blsMultiSigner) Create(pubKeys []string, index uint16) (crypto.MultiSigner, error) {
-	bms.mutSigData.RLock()
-	privKey := bms.data.privKey
-	bms.mutSigData.RUnlock()
-
-	return NewBLSMultisig(bms.llSigner, pubKeys, privKey, bms.keyGen, index)
-}
-
 // CreateSignatureShare returns a BLS single signature over the message
-func (bms *blsMultiSigner) CreateSignatureShare(message []byte, _ []byte) ([]byte, error) {
-	bms.mutSigData.Lock()
-	defer bms.mutSigData.Unlock()
-
-	data := bms.data
-	sigShareBytes, err := bms.llSigner.SignShare(data.privKey, message)
+func (bms *blsMultiSigner) CreateSignatureShare(privateKeyBytes []byte, message []byte) ([]byte, error) {
+	privateKey, err := convertBytesToPrivateKey(privateKeyBytes, bms.keyGen)
 	if err != nil {
 		return nil, err
 	}
 
-	data.sigShares[data.ownIndex] = sigShareBytes
-
-	return sigShareBytes, nil
-}
-
-// not concurrent safe, should be used under RLock mutex
-func (bms *blsMultiSigner) isIndexInBitmap(index uint16, bitmap []byte) error {
-	indexOutOfBounds := index >= uint16(len(bms.data.pubKeys))
-	if indexOutOfBounds {
-		return crypto.ErrIndexOutOfBounds
-	}
-
-	indexNotInBitmap := bitmap[index/8]&(1<<uint8(index%8)) == 0
-	if indexNotInBitmap {
-		return crypto.ErrIndexNotSelected
-	}
-
-	return nil
+	return bms.llSigner.SignShare(privateKey, message)
 }
 
 // VerifySignatureShare verifies the single signature share of the signer with specified position
 // Signature is verified over a message configured with a previous call of SetMessage
```

Comment:
i think the second line of the comment is not relevant anymore, at least regarding `SetMessage`

Reply:
updated the comments